includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -198,12 +198,69 @@ private function check_stable_tag( Check_Result $result, string $readme_file, Pa
 		$warnings = $parser->warnings ? $parser->warnings : array();
 
+		// This should be ERROR rather than WARNING. So ignoring here to handle separately.
+		if ( isset( $warnings['invalid_plugin_name_header'] ) ) {
+			unset( $warnings['invalid_plugin_name_header'] );
+		}
+
 		$warning_keys = array_keys( $warnings );
 
+		$latest_wordpress_version = $this->get_wordpress_stable_version();
+
 		$ignored_warnings = array(
 			'contributor_ignored',
 		);
 
+		$messages = array(
+			'contributor_ignored'          => sprintf(
+				/* translators: %s: plugin header tag */
+				__( 'One or more contributors listed were ignored. The %s field should only contain WordPress.org usernames. Remember that usernames are case-sensitive.', 'plugin-check' ),
+				"'Contributors'"
+			),
+			'requires_php_header_ignored'  => sprintf(
+				/* translators: 1: plugin header tag; 2: Example version 5.2.4. 3: Example version 7.0. */
+				__( 'The %1$s field was ignored. This field should only contain a PHP version such as %2$s or %3$s.', 'plugin-check' ),
+				"'Requires PHP'",
+				"'5.2.4'",
+				"'7.0'"
+			),
+			'tested_header_ignored'        => sprintf(
+				/* translators: 1: plugin header tag; 2: Example version 5.0. 3: Example version 5.1. */
+				__( 'The %1$s field was ignored. This field should only contain a valid WordPress version such as %2$s or %3$s.', 'plugin-check' ),
+				"'Tested up to'",
+				"'" . number_format( floatval( $latest_wordpress_version ), 1 ) . "'",
+				"'" . number_format( floatval( $latest_wordpress_version ) + 0.1, 1 ) . "'"
+			),
+			'requires_header_ignored'      => sprintf(
+				/* translators: 1: plugin header tag; 2: Example version 5.0. 3: Example version 4.9. */
+				__( 'The %1$s field was ignored. This field should only contain a valid WordPress version such as %2$s or %3$s.', 'plugin-check' ),
+				"'Requires at least'",
+				"'" . number_format( floatval( $latest_wordpress_version ), 1 ) . "'",
+				"'" . number_format( floatval( $latest_wordpress_version ) - 0.1, 1 ) . "'"
+			),
+			'too_many_tags'                => sprintf(
+				/* translators: %d: maximum tags limit */
+				__( 'One or more tags were ignored. Please limit your plugin to %d tags.', 'plugin-check' ),
+				5
+			),
+			'ignored_tags'                 => sprintf(
+				/* translators: %s: list of tags not supported */
+				__( 'One or more tags were ignored. The following tags are not permitted: %s', 'plugin-check' ),
+				"'" . implode( "', '", $parser->ignore_tags ) . "'"
+			),
+			'no_short_description_present' => sprintf(
+				/* translators: %s: section title */
+				__( 'The %s section is missing. An excerpt was generated from your main plugin description.', 'plugin-check' ),
+				"'Short Description'"
+			),
+			'trimmed_short_description'    => sprintf(
+				/* translators: 1: section title; 2 maximum limit */
+				__( 'The %1$s section is too long and was truncated. A maximum of %2$d characters is supported.', 'plugin-check' ),
```

Comment:
Mhm I think we'd want to use `_n()` here because the number of characters can be singular or plural.

```suggestion
				_n( 'The %1$s section is too long and was truncated. A maximum of %2$d character is supported.', 'The %1$s section is too long and was truncated. A maximum of %2$d characters is supported.', 150, 'plugin-check' ),
```